src/server/services/Recaptcha.ts

```diff
@@ -0,0 +1,31 @@
+import axios from "axios";
+import { URLSearchParams } from "url";
+
+import { logger } from "../../logger";
+import { config } from "../config";
+import errorCounter from "./ErrorCounter";
+
+export class Recaptcha {
+  constructor(private secret: string = config.Get("RECAPTCHA_SECRET")) {}
+
+  async validate(captcha: string): Promise<boolean> {
+    if (!this.secret) {
```

Comment:
i guess ideally it should be checked during startup? so it's not running app, if the EXTERNAL === true and no recaptcha secret. Could be here too, but if you run application, all green 🌈 but users start sending you errors (and in this case it's `critical`s, as they won't be able to operate with core functionality) that's a bit more unfortunate case, compared to if app wouldn't start and would be automatically rolled-back to working version in kube

Reply:
Agree, updated to crash on startup.